packages/main/src/plugin/kubernetes-client.ts

```diff
@@ -1303,4 +1318,283 @@ export class KubernetesClient {
+        } else if (creationSource.controllerType === 'Job') {
+          await this.restartJob(controllerName, ns);
+        }
+      }
+    } catch (error) {
+      telemetryOptions = { error: error };
+      throw this.wrapK8sClientError(error);
+    } finally {
+      this.telemetry.track('kubernetesRestartPod', telemetryOptions);
+    }
+  }
+
+  protected async restartManuallyCreatedPod(name: string, namespace: string, pod: V1Pod): Promise<void> {
+    const coreApi = this.kubeConfig.makeApiClient(CoreV1Api);
+    await coreApi.deleteNamespacedPod(name, namespace);
+
+    const isDeleted = await this.waitForPodDeletion(coreApi, name, namespace);
+    if (!isDeleted) {
+      throw new Error(`pod "${name}" in namespace "${namespace}" was not deleted within the expected timeframe`);
+    }
+
+    delete pod.metadata?.resourceVersion;
+    delete pod.metadata?.uid;
+    delete pod.metadata?.selfLink;
+    delete pod.metadata?.creationTimestamp;
+    delete pod.status;
+
+    const newPod: V1Pod = { ...pod };
+    await coreApi.createNamespacedPod(namespace, newPod);
+  }
+
+  protected async waitForPodDeletion(
+    coreApi: CoreV1Api,
+    name: string,
+    namespace: string,
+    timeout: number = 60000,
+  ): Promise<boolean> {
+    const startTime = Date.now();
+
+    while (Date.now() - startTime < timeout) {
+      try {
+        await coreApi.readNamespacedPodStatus(name, namespace);
+        await new Promise(resolve => setTimeout(resolve, 1000));
+      } catch (e) {
+        const error = e ?? {};
+        if (typeof error === 'object' && 'response' in error) {
+          const axiosError = error as { response: { statusCode: number } };
+          if (axiosError.response.statusCode === 404) {
+            return true;
+          }
+        }
+        throw e;
+      }
+    }
+
+    return false;
+  }
+
+  protected async scaleControllerToRestartPods(
+    namespace: string,
+    controllerName: string,
+    controllerType: ScalableControllerType,
+    timeout: number = 10000,
+  ): Promise<void> {
+    const appsApi = this.kubeConfig.makeApiClient(AppsV1Api);
+
+    let currentReplicas = 0;
+    if (controllerType === 'Deployment') {
+      const { body: currentDeployment } = await appsApi.readNamespacedDeployment(controllerName, namespace);
+      currentReplicas = currentDeployment.spec?.replicas || 1;
+    } else if (controllerType === 'ReplicaSet') {
+      const { body: currentReplicaSet } = await appsApi.readNamespacedReplicaSet(controllerName, namespace);
+      currentReplicas = currentReplicaSet.spec?.replicas || 1;
+    } else if (controllerType === 'StatefulSet') {
+      const { body: currentStatefulSet } = await appsApi.readNamespacedStatefulSet(controllerName, namespace);
+      currentReplicas = currentStatefulSet.spec?.replicas || 1;
+    }
+
+    await this.scaleController(appsApi, namespace, controllerName, controllerType, 0);
```

Comment:
This works for scalable resource with only 1 replica, right? If there are more than 1 replica, all will be killed, but this is not what we want

Reply:
Yes, it should scale the replica to 0 and rescale to the original number. I think, we can provide enhancement for this in the next PR as the next step.